lib/src/middleware.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:async';
+
+import 'client.dart';
+import 'handler_client.dart';
+import 'request.dart';
+import 'response.dart';
+
+typedef Client Middleware(Client inner);
+
+typedef Future<Response> ResponseHandler(Response response);
+typedef Future<Request> RequestHandler(Request request);
+
+Middleware createMiddleware({
+  RequestHandler requestHandler,
+  ResponseHandler responseHandler,
```

Comment:
Not right now. We may find a compelling use-case at some point, but I think it's a lot less crucial on the client since you aren't trying to mount a WebSocket handler as part of an existing app or whatever.